tests/integration/test_charm.py

```diff
@@ -543,3 +543,92 @@ async def test_anonymize_user(
             timeout=5,
         )
     assert res.status_code == 403
+
+
+@pytest.mark.usefixtures("s3_backup_bucket")
+async def test_synapse_enable_s3_backup_integration_success(
+    model: Model,
+    synapse_app: Application,
+    s3_backup_configuration: dict,
+):
+    """
+    arrange: Synapse App deployed and s3-integrator deployed with bucket created.
+    act:  integrate s3-integrator with Synapse.
+    assert: Synapse gets into active status.
+    """
+    s3_integrator_app = await model.deploy(
+        "s3-integrator",
+        channel="latest/edge",
+        config={
+            "endpoint": s3_backup_configuration["endpoint"],
+            "bucket": s3_backup_configuration["bucket"],
+            "path": s3_backup_configuration["path"],
+            "region": s3_backup_configuration["region"],
+            "s3-uri-style": s3_backup_configuration["s3-uri-style"],
+        },
+    )
+    await model.wait_for_idle(apps=[s3_integrator_app.name], idle_period=5, status="blocked")
+    action_sync_s3_credentials: Action = await s3_integrator_app.units[0].run_action(
+        "sync-s3-credentials",
+        **{
```

Comment:
Why unpack when these could just be arguments?

Reply:
Refactored a bit to get the credentials from a fixture. Unfortunately, the arguments in this case contain `-`, so they are not valid Python identifiers, and it is not possible to use them as arguments.